cmd/snap-confine/snap-confine.c

```diff
@@ -286,6 +287,14 @@ static void sc_restore_process_state(const sc_preserved_process_state *
 	debug("the process has been placed in the special void directory");
 }
 
+static void log_startup_phase(const char *stage)
+{
+	struct timeval tv;
+	gettimeofday(&tv, NULL);
+	debug("-- snap startup {\"stage\":\"%s\", \"time\":\"%lu.%06lu\"}",
+	      stage, tv.tv_sec, tv.tv_usec);
```

Comment:
micro-optimisation nitpick: I would wrap this in a `sc_is_debug_enabled()` to avoid calling `gettimeofday()` when not needed, or even turning `log_startup_phase()` into a macro.

Reply:
gettimeofday() is heavily optimized for and should be called through vDSO as one of the most often called syscalls, but I can wrap the whole thing with a check whether debugs is enabled.